sprout/core/edit_package_properties.py

```diff
@@ -48,10 +49,35 @@ def edit_package_properties(path: Path, properties: dict) -> dict:
         JSONDecodeError: If the `datapackage.json` file couldn't be read.
     """
     verify_is_file(path)
-    verify_package_properties(properties)
 
+    # verify current properties
     current_properties = read_json(path)
     verify_properties_are_well_formed(current_properties, PackageError.type)
 
-    current_properties.update(properties)
-    return current_properties
+    # verify new properties + remove default values to avoid overwriting existing values
+    verify_properties_are_well_formed(properties, PackageError.type)
+    properties = remove_default_package_properties_from_dict(properties)
+
+    return verify_package_properties(current_properties | properties)
```

Comment:
I'm a bit confused about what the `|` does here, could you explain that? Reading about `|`, it seems like it compares bits between objects. What does it do here?

Reply:
They describe it in [this post](https://ridwanray.medium.com/the-pipe-symbol-in-python-133239503fec).
As of my understanding both `|` and `.update()` "merges" or "unites" dictionaries and if a key is in both the dictionaries, the value from the second dictionary is used. The main difference is that the `|` returns a new dictionary while `.update()`edits the first dictionary and returns `None`.

So I could accomplish the same by writing this instead: 

```suggestion
	current_properties.update(properties)
    return verify_package_properties(current_properties)
``` 

So, the bug was not because of `.update()`, but rather because the `properties` dict included default values that was overwritten. With a `properties` dict with default values, the `|` would also overwrite the values in `current_properties`